dev/src/gltfwrapper.cpp

```diff
@@ -0,0 +1,81 @@
+// This is free and unencumbered software released into the public domain.
+//
+// Anyone is free to copy, modify, publish, use, compile, sell, or
+// distribute this software, either in source code form or as a compiled
+// binary, for any purpose, commercial or non-commercial, and by any
+// means.
+//
+// In jurisdictions that recognize copyright laws, the author or authors
+// of this software dedicate any and all copyright interest in the
+// software to the public domain. We make this dedication for the benefit
+// of the public at large and to the detriment of our heirs and
+// successors. We intend this dedication to be an overt act of
+// relinquishment in perpetuity of all present and future rights to this
+// software under copyright law.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+// OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+#include <flatbuffers/idl.h>
+#include <stb/stb_image.h>
+
+#include "lobster/stdafx.h"
+
+#include "lobster/natreg.h"
+
+#include "lobster/glinterface.h"
+
+#include "lobster/engine.h"
+
+#include "lobster/flat_gltf.h"
+
+using namespace lobster;
+
+typedef fgltf::Scene Scene;
+
+ResourceType scene_type = {"scene", [](void *s) { delete (Scene *)s; }};
+
+Scene &GetScene(VM &vm, Value &res) {
+  return *GetResourceDec<Scene *>(vm, res, &scene_type);
+}
+
+Scene *LoadGLTF(string_view filename) {
+  Scene *scene = new fgltf::Scene;
+  // TODO: accepting a string_view only to convert it to a string is ass
+  // but that's how you combine Lobster's coding style with a more traditional
+  // coding style
+  *scene = fgltf::LoadScene(std::string(filename), "schemas/gltf.fbs");
+  if (scene->roots.size() == 0) {
+    delete scene;
+    return nullptr;
+  }
+  return scene;
+}
+
+// found in graphics.cpp
+extern void TestGL(VM &vm);
+
+void AddGLTF(NativeRegistry &nfr) {
+
+  nfr("gl_new_scene_gltf", "filename", "S", "R?",
+      "load a .gltf file into a scene, returns scene or nil on failure to "
+      "load.",
+      [](VM &vm, Value &fn) {
+        TestGL(vm);
+        auto s = LoadGLTF(fn.sval()->strv());
+        return s ? Value(vm.NewResource(s, &scene_type)) : Value();
+      });
+
+  nfr("gl_render_scene", "s", "R", "", "renders the specified scene",
```

Comment:
lets have both functions have the world `gltf` in them somewhere for clarity.

Reply:
`gl_load_scene_gltf` sorta sounds alright, but should it be something like `gltf_load_scene` and `gltf_render_scene` or `gl_load_scene_gltf` and `gl_render_scene_gltf` or `gl_load_gltf_scene` and `gl_render_gltf_scene`? I'd like to keep the word `scene` in there since as soon as the model is packed into the FlatBuffers format it's technically not a glTF-compliant file, but it's still a `scene`.

Reply:
Either way works for me.